### DISCUSSION TITLE
Native functions with yaml file

### DISCUSSION THREAD

**Discussion by bandara1987** [microsoft/semantic-kernel#6644] [Q&A] [answered]
I know we can load functions with yaml file sin SK. is there way to load native functions with yaml files as well? the reason is in native functions function/parameter descriptions are hard coded. i need a way to load these from a configuration.

for an example, i need to implement below function in a way that descriptions "Validates the Task Id provided by the user" and "Task Id provided by the user" are coming from a configuration. not hardcoded in the class.

```
[KernelFunction("ValidateTaskId")]
[Description("Validates the Task Id provided by the user")]
public string ValidateTaskId(Kernel kernal, [Description("Task Id provided by the user"), Required] string taskId)
{
    if (taskId.Equals("1234"))
        return "Valid task id";
    else
        return "Invalid task id";
}
```

**Comment by markwallace-microsoft** [comment, marked as answer]
@bandara1987 Take a look at the sample here https://github.com/microsoft/semantic-kernel/pull/6650/files.

This is one approach you can use to load the function metadata from a YAML file.

**Reply by bandara1987** [reply]
Thanks a lot. works perfectly :)
